mesonpy/__init__.py

```diff
@@ -903,6 +907,12 @@ def _info(self, name: str) -> Dict[str, Any]:
 
     @property
     def _install_plan(self) -> Dict[str, Dict[str, Dict[str, str]]]:
+        install_plan = self._info('intro-install_plan').copy()
+
+        for files in install_plan.values():
+            for file, details in list(files.items()):
+                if details['tag'] not in self._meson_args['install-tags']:
+                    del files[file]
```

Comment:
I'm not sure if we should do it here.

Reply:
i updated this block to parse the "install" args instead of using an extra 'install-tags' field
```suggestion
    def _install_plan(self) -> Dict[str, Dict[str, Dict[str, str]]]:
        """Meson install_plan metadata."""
    
        # copy the install plan so we can modify it
        install_plan = self._info('intro-install_plan').copy()

        # parse install args for install tags (--tags)
        install_tags = []
        for arg in self._meson_args['install']:
            if arg.strip().startswith('--tags='):
                install_tags = arg.split('=', 1)[1].split(',')
                break
        else:
            return install_plan
            
        # filter out files that do not fit the install tags
        for files in install_plan.values():
            for file, details in list(files.items()):
                if details['tag'].strip() not in install_tags:
                    del files[file]

        return install_plan
```

Reply:
@FFY00 This is just a suggestion (it was easy to implement it here). 
If you give me some hints or thoughts on your doubts I could look for a better place to put this filter.

Reply:
This is a bit fragile, and I'd prefer not having to maintain it. I think it should be pretty easy to use `argparse`, as I mentioned in https://github.com/mesonbuild/meson-python/pull/267#discussion_r1072979001. What do you think?

Reply:
Oh ja, I didn't see the comment 🙂 I'll update the suggestion tomorrow

Reply:
No worries. Thank you :blush:

Reply:
Adapted suggestion

```suggestion
    """Meson install_plan metadata."""
    
        # copy the install plan so we can modify it
        install_plan = self._info('intro-install_plan').copy()

        # parse install args for install tags (--tags)
        parser = argparse.ArgumentParser()
        parser.add_argument('--tags')
        args, _ = parser.parse_known_args(self._meson_args['install'])
        
        # filter the install_plan for files that do not fit the install tags
        if args.tags:
            install_tags = args.tags.split(',')
                
            for files in install_plan.values():
                for file, details in list(files.items()):
                    if details['tag'].strip() not in install_tags:
                        del files[file]

        return install_plan
```

naturally this needs `import argparse` at the beginning